app/models/vm_or_template/operations/configuration.rb

```diff
@@ -73,8 +73,8 @@ def disconnect_floppies
 
   def raw_add_disk(disk_name, disk_size_mb, options = {})
     raise _("VM has no EMS, unable to add disk") unless ext_management_system
-    if options[:datastore]
-      datastore = Storage.find_by(:name => options[:datastore])
+    if options[:datastore] || options[:datastore_id]
```

Comment:
@durandom the use case in which datastore appears twice for the same RHV provider happened when the provider was first added to manageiq, then removed and added again.

The previous datastore remained on cmdb, however, its association with any provider was removed. Therefore the code above will allow to select only a datastore which is associated with a datastore related to the provider in which the vm is about to be added.

However, @agrare raised an issue on https://github.com/ManageIQ/manageiq-providers-ovirt/pull/42 where VMware might have several datastores with the same name.